src/main/java/decodes/tsdb/algo/AverageAlgorithm.java

```diff
@@ -1,84 +1,71 @@
 package decodes.tsdb.algo;
 
-import java.util.Date;
-
-import ilex.var.NamedVariableList;
 import ilex.var.NamedVariable;
-import decodes.tsdb.DbAlgorithmExecutive;
 import decodes.tsdb.DbCompException;
-import decodes.tsdb.DbIoException;
-import decodes.tsdb.VarFlags;
 import decodes.util.PropertySpec;
+import org.opendcs.annotations.algorithm.Algorithm;
+import org.opendcs.annotations.algorithm.Input;
+import org.opendcs.annotations.algorithm.Output;
 
-//AW:JAVADOC
-/**
-AverageAlgorithm averages single 'input' parameter to a single 'average' 
-parameter. The averaging period is determined by the interval of the output
-parameter.
+@Algorithm(
+		description = "AverageAlgorithm averages single 'input' parameter to a single 'average' \n" +
+				"parameter. The averaging period is determined by the interval of the output\n" +
+				"parameter."
+)
 
- */
-//AW:JAVADOC_END
 public class AverageAlgorithm extends decodes.tsdb.algo.AW_AlgorithmBase
 {
-//AW:INPUTS
-	double input;	//AW:TYPECODE=i
-	String _inputNames[] = { "input" };
-//AW:INPUTS_END
+	private static final String AVERAGESTRING = "average";
+	@Input
+	double input;
 
-//AW:LOCALVARS
 	double tally;
 	int count;
-	PropertySpec specs[] = 
+	PropertySpec specs[] =
 	{
 		new PropertySpec("negativeReplacement", PropertySpec.NUMBER, 
 			"(no default) If set, and output would be negative, then replace with the number supplied.")
 	};
+	@Override
+	protected PropertySpec[] getAlgoPropertySpecs()
+	{
+		return specs;
+	}
 
-//AW:LOCALVARS_END
+	public double negativeReplacement = Double.NEGATIVE_INFINITY;
```

Comment:
Put the PropertySpec annotation on this one.

Reply:
@MikeNeilson - how should I handle the String value? Set it as "Double.NEGATIVE_INFINITY" or "-1 / 0" ?

Reply:
Without any value would work, I think, but probably better to just copy the actual value in. You can get the actual value from the java source or something.

Going to have to think about that though, long term should probably be the text "missing" in case some implementation doesn't want to use NEGATIVE_INFINITY to show that, but we don't need to decided that now.